diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5a4309c..0ecbf08 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,6 +27,10 @@ project(absl LANGUAGES CXX VERSION 20260817)
 set(ABSL_SOVERSION "2608.0.0")
 include(CTest)
 
+if(MINGW)
+  add_compile_definitions(____FIReference_1_boolean_INTERFACE_DEFINED__)
+endif()
+
 # Output directory is correct by default for most build setups. However, when
 # building Abseil as a DLL, it is important to have the DLL in the same
 # directory as the executable using it. Thus, we put all executables in a single
