--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -1748,9 +1748,16 @@
         find_dependency(ONNX)\n\
         find_dependency(re2)\n\
         find_dependency(flatbuffers)\n\
+        if(NOT APPLE AND NOT CMAKE_SYSTEM_NAME STREQUAL \"Emscripten\")\n\
         find_dependency(cpuinfo)\n\
-        find_dependency(protobuf)\n\
-        find_dependency(Boost COMPONENTS mp11)\n\
+        endif()\n\
+        find_dependency(protobuf)\n")
+  if(onnxruntime_USE_KLEIDIAI)
+    string(APPEND PROJECT_CONFIG_CONTENT
+        "find_dependency(KleidiAI)\n")
+  endif()
+  string(APPEND PROJECT_CONFIG_CONTENT
+        "        find_dependency(Boost COMPONENTS mp11)\n\
         find_dependency(Microsoft.GSL 4.0)\n\
         if(NOT WIN32 AND NOT CMAKE_SYSTEM_NAME STREQUAL \"Android\")\n\
         find_dependency(Iconv)\n\
