diff -ruN a/cmake/onnxruntime_mlas.cmake b/cmake/onnxruntime_mlas.cmake
--- a/cmake/onnxruntime_mlas.cmake	2025-10-22 01:09:33.000000000 +0200
+++ b/cmake/onnxruntime_mlas.cmake	2026-08-23 12:23:46.108661015 +0200
@@ -268,23 +268,14 @@
 endfunction()
 
 function(setup_kleidiai)
+  find_package(KleidiAI CONFIG REQUIRED)
   target_sources(onnxruntime_mlas PRIVATE
-    ${MLAS_SRC_DIR}/kai_ukernel_interface.cpp
-    ${MLAS_SRC_DIR}/kleidiai/sgemm_kleidiai.cpp
-    ${MLAS_SRC_DIR}/kleidiai/convolve_kleidiai.cpp
-    ${MLAS_SRC_DIR}/kleidiai/qgemm_kleidiai.cpp
+    /kai_ukernel_interface.cpp
+    /kleidiai/sgemm_kleidiai.cpp
+    /kleidiai/convolve_kleidiai.cpp
+    /kleidiai/qgemm_kleidiai.cpp
   )
-  target_link_libraries(onnxruntime_mlas PRIVATE kleidiai)
-  list(APPEND onnxruntime_EXTERNAL_LIBRARIES kleidiai)
-  set(onnxruntime_EXTERNAL_LIBRARIES ${onnxruntime_EXTERNAL_LIBRARIES} PARENT_SCOPE)
-
-  if (NOT onnxruntime_BUILD_SHARED_LIB)
-    install(TARGETS kleidiai EXPORT ${PROJECT_NAME}Targets
-    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-    FRAMEWORK DESTINATION ${CMAKE_INSTALL_BINDIR})
-  endif()
+  target_link_libraries(onnxruntime_mlas PRIVATE KleidiAI::kleidiai)
 endfunction()
 
 if (CMAKE_SYSTEM_NAME STREQUAL "Emscripten")
diff -ruN a/onnxruntime/core/mlas/lib/mlasi.h b/onnxruntime/core/mlas/lib/mlasi.h
--- a/onnxruntime/core/mlas/lib/mlasi.h	2025-10-22 01:09:33.000000000 +0200
+++ b/onnxruntime/core/mlas/lib/mlasi.h	2026-08-23 12:23:46.108758080 +0200
@@ -2707,8 +2707,12 @@
 MlasReadTimeStampCounter(void)
 {
 #ifdef _WIN32
-#if defined(MLAS_TARGET_AMD64_IX86)
+#if defined(_MSC_VER) && defined(MLAS_TARGET_AMD64_IX86)
     return ReadTimeStampCounter();
+#elif defined(__GNUC__) && (defined(MLAS_TARGET_AMD64) || defined(MLAS_TARGET_IX86))
+    uint32_t eax, edx;
+    __asm__ __volatile__ ("rdtsc" : "=a" (eax), "=d" (edx));
+    return ((uint64_t)edx << 32) | eax;
 #else
     LARGE_INTEGER PerformanceCounter;
 
diff -ruN a/onnxruntime/core/mlas/lib/x86/SgemmKernelAvx.S b/onnxruntime/core/mlas/lib/x86/SgemmKernelAvx.S
--- a/onnxruntime/core/mlas/lib/x86/SgemmKernelAvx.S	2025-10-22 01:09:33.000000000 +0200
+++ b/onnxruntime/core/mlas/lib/x86/SgemmKernelAvx.S	2026-08-23 12:23:46.108789459 +0200
@@ -342,8 +342,12 @@
 
 .LOutputMasked8x2Block:
         neg     ebp
+#if defined(_WIN32)
+        lea     ebx,DWORD PTR C_UNDERSCORE(MlasMaskMoveTableAvx)
+#else
         LoadGlobalOffsetTable bx
         mov     ebx,DWORD PTR C_UNDERSCORE(MlasMaskMoveTableAvx)@GOT[ebx]
+#endif
         vmovdqu ymm0,YMMWORD PTR [ebx+ebp*4+8*4]
         cmp     BYTE PTR .LSgemmKernelFrame_ZeroMode[esp],0
         jnz     .LSkipAccumulateMasked8x2Block
@@ -420,8 +424,12 @@
 
 .LOutputMasked8x1Block:
         neg     ebp
+#if defined(_WIN32)
+        lea     ebx,DWORD PTR C_UNDERSCORE(MlasMaskMoveTableAvx)
+#else
         LoadGlobalOffsetTable bx
         mov     ebx,DWORD PTR C_UNDERSCORE(MlasMaskMoveTableAvx)@GOT[ebx]
+#endif
         vmovdqu ymm0,YMMWORD PTR [ebx+ebp*4+8*4]
         cmp     BYTE PTR .LSgemmKernelFrame_ZeroMode[esp],0
         jnz     .LSkipAccumulateMasked8x1Block
