--- a/cmake/adjust_global_compile_flags.cmake
+++ b/cmake/adjust_global_compile_flags.cmake
@@ -45,7 +45,7 @@
     string(APPEND CMAKE_CXX_FLAGS " -msimd128")
   endif()
 
-  if (onnxruntime_ENABLE_WEBASSEMBLY_EXCEPTION_CATCHING)
+  if (onnxruntime_ENABLE_WEBASSEMBLY_EXCEPTION_CATCHING AND NOT CMAKE_CXX_FLAGS MATCHES "-fwasm-exceptions")
     string(APPEND CMAKE_C_FLAGS " -s DISABLE_EXCEPTION_CATCHING=0")
     string(APPEND CMAKE_CXX_FLAGS " -s DISABLE_EXCEPTION_CATCHING=0")
   endif()
--- a/cmake/external/abseil-cpp.cmake
+++ b/cmake/external/abseil-cpp.cmake
@@ -122,7 +122,7 @@
 absl::hash_function_defaults
 absl::function_ref
 absl::city
-# absl::low_level_hash # abseil 20250814.1+
+# # absl::low_level_hash # abseil 20250814.1+
 absl::fixed_array
 absl::variant
 absl::meta
--- a/cmake/onnxruntime.cmake
+++ b/cmake/onnxruntime.cmake
@@ -76,7 +76,7 @@
 
   add_custom_target(onnxruntime_generate_def ALL DEPENDS ${SYMBOL_FILE} ${CMAKE_CURRENT_BINARY_DIR}/generated_source.c)
   endif()
-  if(WIN32)
+  if(WIN32 AND onnxruntime_BUILD_SHARED_LIB)
     onnxruntime_add_shared_library(onnxruntime
       ${SYMBOL_FILE}
       "${ONNXRUNTIME_ROOT}/core/dll/dllmain.cc"
@@ -279,7 +279,7 @@
   )
 endif()
 
-if(WIN32)
+if(WIN32 AND onnxruntime_BUILD_SHARED_LIB)
   target_link_options(onnxruntime PRIVATE ${onnxruntime_DELAYLOAD_FLAGS})
 endif()
 #See: https://cmake.org/cmake/help/latest/prop_tgt/SOVERSION.html
@@ -312,10 +312,11 @@
         ARCHIVE   DESTINATION ${CMAKE_INSTALL_LIBDIR}
         LIBRARY   DESTINATION ${CMAKE_INSTALL_LIBDIR}
         RUNTIME   DESTINATION ${CMAKE_INSTALL_BINDIR}
-        FRAMEWORK DESTINATION ${CMAKE_INSTALL_BINDIR})
+        FRAMEWORK DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(FILES ${ONNXRUNTIME_PUBLIC_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/onnxruntime)
 
 
-if (WIN32 AND NOT CMAKE_CXX_STANDARD_LIBRARIES MATCHES kernel32.lib)
+if (WIN32 AND onnxruntime_BUILD_SHARED_LIB AND NOT CMAKE_CXX_STANDARD_LIBRARIES MATCHES kernel32.lib)
   # Workaround STL bug https://github.com/microsoft/STL/issues/434#issuecomment-921321254
   # Note that the workaround makes std::system_error crash before Windows 10
 
--- a/cmake/onnxruntime_mlas.cmake
+++ b/cmake/onnxruntime_mlas.cmake
@@ -782,7 +782,7 @@
 endif()
 
 if (NOT onnxruntime_BUILD_SHARED_LIB)
-    install(TARGETS onnxruntime_mlas EXPORT ${PROJECT_NAME}Targets
+    install(TARGETS ${ONNXRUNTIME_MLAS_LIBS} EXPORT ${PROJECT_NAME}Targets
             ARCHIVE   DESTINATION ${CMAKE_INSTALL_LIBDIR}
             LIBRARY   DESTINATION ${CMAKE_INSTALL_LIBDIR}
             RUNTIME   DESTINATION ${CMAKE_INSTALL_BINDIR}
--- a/cmake/onnxruntime_webassembly.cmake
+++ b/cmake/onnxruntime_webassembly.cmake
@@ -94,7 +94,7 @@
   )
 
   # Override re2 compiler options to remove -pthread
-  # set_property(TARGET re2 PROPERTY COMPILE_OPTIONS )
+  # # set_property(TARGET re2 PROPERTY COMPILE_OPTIONS )
 endif()
 
 if (NOT onnxruntime_USE_VCPKG)
@@ -305,7 +305,7 @@
           #       However, it may be very slow.
           # "SHELL:-s ASSERTIONS=2"
           "SHELL:-s ASSERTIONS=1"
-          "SHELL:-s SAFE_HEAP=1"
+          "SHELL:-s SAFE_HEAP=0"
           "SHELL:-s STACK_OVERFLOW_CHECK=2"
         )
     endif()
@@ -326,7 +326,9 @@
   endif()
 
   # Set link flag to enable exceptions support, this will override default disabling exception throwing behavior when disable exceptions.
-  target_link_options(onnxruntime_webassembly PRIVATE "SHELL:-s DISABLE_EXCEPTION_THROWING=0")
+  if (NOT CMAKE_CXX_FLAGS MATCHES "-fwasm-exceptions")
+    target_link_options(onnxruntime_webassembly PRIVATE "SHELL:-s DISABLE_EXCEPTION_THROWING=0")
+  endif()
 
   if (onnxruntime_ENABLE_WEBASSEMBLY_PROFILING)
     target_link_options(onnxruntime_webassembly PRIVATE --profiling --profiling-funcs)
--- a/onnxruntime/core/session/utils.cc
+++ b/onnxruntime/core/session/utils.cc
@@ -415,7 +415,7 @@
                                                             true,
                                                             ProviderLibraryPathType::Absolute);
   bool is_provider_bridge = provider_library->Load() == Status::OK();  // library has GetProvider
-  LOGS_DEFAULT(INFO) << "Loading EP library: " << library_path
+  LOGS_DEFAULT(INFO) << "Loading EP library: " << resolved_library_path
                      << (is_provider_bridge ? " as a provider bridge" : " as a plugin");
 
   // create EpLibraryPlugin to ensure CreateEpFactories and ReleaseEpFactory are available
